rcl/test/rcl/test_wait.cpp

```diff
@@ -269,15 +273,16 @@ TEST(CLASSNAME(WaitSetTestFixture, RMW_IMPLEMENTATION), multi_wait_set_threaded)
   //   ss << "]";
   //   printf("%s\n", ss.str().c_str());
   // };
+
   size_t loop_count = 0;
   while (loop_test()) {
     loop_count++;
-    // print_state("triggering, current states: ");
     for (const auto & test_set : test_sets) {
       ret = rcl_trigger_guard_condition(&test_set.guard_condition);
       EXPECT_EQ(RCL_RET_OK, ret) << rcl_get_error_string_safe();
     }
-    std::this_thread::sleep_for(trigger_period);
+    std::this_thread::yield();
```

Comment:
That answer is talking about the fact that the OP is using a different method to compare to sleep_for and the accuracies might be different. However, the reference website (and likely the standard) says that `sleep_for` uses a steady clock to gage the duration and is very specific that it should sleep for *at least* the requested period.

From http://en.cppreference.com/w/cpp/thread/sleep_for:
> Blocks the execution of the current thread for at least the specified sleep_duration.
A steady clock is used to measure the duration. This function may block for longer than sleep_duration due to scheduling or resource contention delays.

The SO link you sent also points out that the `sleep_for` function on Windows is already a loop around `Sleep` function to handle spurious wake ups.